Code/Framework/AzToolsFramework/AzToolsFramework/Prefab/Instance/InstanceUpdateExecutor.cpp

```diff
@@ -94,9 +121,108 @@ namespace AzToolsFramework
         void InstanceUpdateExecutor::RemoveTemplateInstanceFromQueue(const Instance* instance)
         {
             AZStd::erase_if(m_instancesUpdateQueue, [instance](Instance* entry)
+                {
+                    return entry == instance;
+                }
+            );
+        }
+
+        const void InstanceUpdateExecutor::ReplaceFocusedContainerTransformAccordingToRoot(
+            const Instance* focusedInstance, PrefabDom& focusedInstanceDom) const
+        {
+            // Climb from the focused instance to the root and store the path.
+            AZStd::string rootToFocusedInstance;
+            auto rootInstance = PrefabDomUtils::ClimbUpToTargetInstance(focusedInstance, nullptr, rootToFocusedInstance);
+
+            if (rootInstance != focusedInstance)
+            {
+                // Create the path from the root instance to the container entity of the focused instance.
+                AZStd::string rootToFocusedInstanceContainer =
+                    AZStd::string::format("%s/%s", rootToFocusedInstance.c_str(), PrefabDomUtils::ContainerEntityName);
+                PrefabDomPath rootToFocusedInstanceContainerPath(rootToFocusedInstanceContainer.c_str());
+
+                // Retrieve the dom of the root instance.
+                PrefabDom rootDom;
+                rootDom.CopyFrom(
+                    m_prefabSystemComponentInterface->FindTemplateDom(rootInstance->GetTemplateId()), focusedInstanceDom.GetAllocator());
+
+                PrefabDom containerDom;
+                containerDom.CopyFrom(*rootToFocusedInstanceContainerPath.Get(rootDom), focusedInstanceDom.GetAllocator());
+
+                // Paste the focused instance container dom as seen from the root into instanceDom.
+                AZStd::string containerName =
+                    AZStd::string::format("/%s", PrefabDomUtils::ContainerEntityName);
+                PrefabDomPath containerPath(containerName.c_str());
+                containerPath.Set(focusedInstanceDom, containerDom, focusedInstanceDom.GetAllocator());
+            }
+        }
+
+        bool InstanceUpdateExecutor::GenerateInstanceDomAccordingToCurrentFocus(const Instance* instance, PrefabDom& instanceDom)
```

Comment:
This feels like the InstanceUpdateExecutor is trying to too much. Classes should follow single-responsibility-principle to keep systems maintainable. The responsibility of this class should be, "given a queue of instances, I'll propagate given Doms to them". It doesn't even need to know what a focus mode is. So I recommend you create another class that outputs a Dom to be used for propagation by taking an instance as input. The class (may be InstanceDomGenerator) can do focus specific logic and fetch the required dom.